core/dbt/parser/manifest.py

```diff
@@ -207,13 +210,14 @@ def load(self):
             saved_files = self.saved_manifest.files
         for project in self.all_projects.values():
             read_files(project, self.manifest.files, project_parser_files, saved_files)
+        orig_project_parser_files = project_parser_files
         self._perf_info.path_count = len(self.manifest.files)
         self._perf_info.read_files_elapsed = (time.perf_counter() - start_read_files)
 
         skip_parsing = False
         if self.saved_manifest is not None:
-            partial_parsing = PartialParsing(self.saved_manifest, self.manifest.files)
-            skip_parsing = partial_parsing.skip_parsing()
+            self.partial_parser = PartialParsing(self.saved_manifest, self.manifest.files)
+            skip_parsing = self.partial_parser.skip_parsing()
             if skip_parsing:
```

Comment:
I really like the way this reads. There's another skip check down below for these new macro overrides. Can we fold that check inside `skip_parsing()`? Reading here, I feel like if `skip_parsing` is `False` we know it is safe to partial parse.